Lib/ssl.py

```diff
@@ -1000,65 +1000,61 @@ def _create(cls, sock, server_side=False, do_handshake_on_connect=True,
         )
         self = cls.__new__(cls, **kwargs)
         super(SSLSocket, self).__init__(**kwargs)
-        sock_timeout = sock.gettimeout()
-        sock.detach()
+        try:
+            sock_timeout = sock.gettimeout()
+            sock.detach()
```

Comment:
I think it would be better to call `sock.gettimeout()` and `sock.detach()` outside the try/catch

```python
        sock_timeout = sock.gettimeout()
        kwargs = dict(
            family=sock.family, type=sock.type, proto=sock.proto,
            fileno=sock.detatch()
        )
        self = cls.__new__(cls, **kwargs)
        super(SSLSocket, self).__init__(**kwargs)
        try:
            self._context = context
            ...
```

Reply:
But what if `__new__` or `__init__` raise?

Reply:
Then they should be responsible for closing the passed in fileno

Reply:
They are not. If you pass a fileno, you are responsible (in this case it is the responsibility of `sock` until `detach()` is called).